### PR TITLE
msg/async/AsyncConnection: combine multi alloc into one

### PR DESCRIPTION
**Pull request by yuyuyu101**
Signed-off-by: Haomai Wang <haomai@xsky.com>

### CODE REVIEW

File: src/msg/async/AsyncConnection.cc [hunk 1]
```diff
@@ -701,7 +699,7 @@ void AsyncConnection::process()
                     << message->get_seq() << " <= " << cur_seq << " " << message << " " << *message
                     << ", discarding" << dendl;
             message->put();
-            if (has_feature(CEPH_FEATURE_RECONNECT_SEQ) && async_msgr->cct->_conf->ms_die_on_old_message)
+            if (jas_feature(CEPH_FEATURE_RECONNECT_SEQ) && async_msgr->cct->_conf->ms_die_on_old_message)
```
Comment (xiexingguo) [inline src/msg/async/AsyncConnection.cc]: changed by accident?

**Comment by yuyuyu101** [comment]
OH, fixed

File: src/msg/async/AsyncConnection.cc [hunk 2]
```diff
@@ -101,22 +101,20 @@ class C_tick_wakeup : public EventCallback {
...
-    data.push_back(buffer::create_page_aligned(middle));
-    left -= middle;
-  }
-  if (left) {
-    data.push_back(buffer::create(left));
-  }
+  alloc_len += left;
+  bufferptr ptr(buffer::create_page_aligned(alloc_len));
+  if (head)
+    ptr.set_offset(CEPH_PAGE_SIZE-head);
```
Comment (tchaikov) [inline src/msg/async/AsyncConnection.cc]: nit, add parentheses around `-`.

Comment (yuyuyu101) [reply src/msg/async/AsyncConnection.cc]: ptr.set_offset(CEPH_PAGE_SIZE - head);
?

Comment (tchaikov) [reply src/msg/async/AsyncConnection.cc]: yes.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-11-13_07:26:34-rados-wip-kefu-testing-2017-11-13-1300-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-11-14_05:25:42-rados-wip-kefu-testing-2017-11-14-1048-distro-basic-mira/

**Comment by tchaikov** [comment]
@yuyuyu101 ping?

**Comment by yuyuyu101** [comment]
done

**Comment by yuriw** [comment]
[wip-yuri2-testing-2018-01-09-1813](https://trello.com/c/x6we0BIB/224-wip-yuri2-testing-2018-01-09-1813)

**Pull request merged**